### ISSUE TITLE
Put Wordpress permalinks on all metric pages

### ISSUE THREAD

**Issue by geekygirldawn** [chaoss/website#781]
As I mentioned in Slack, I used Georg’s idea that we can get the page id from the html source, and I spent a bit of time this morning using BeautifulSoup in python to grab all of the permalinks and map them to the human readable links, github url, and wg so that we can eventually do PRs to put the permalink on the metrics web pages. 

[chaoss_urls.csv](https://github.com/chaoss/website/files/10444252/chaoss_urls.csv)

We need to decide how we want to format those links and where we want to put them on the page before we start, but I thought having an issue for this would help us keep track of the discussion.

We could also probably use this csv file as input to another script that could auto generate the PRs, but that might be a bit more work.

**Comment by GeorgLink** [comment]
> We need to decide how we want to format those links

I like the format you included in the csv file, with the shorter `p` instead of `page_id`. 

Human readable and SEO friendly URL:
* https://chaoss.community/metric-bus-factor/

Static URL for tools and references that will resolve to the human readable URL. The two options are:
* https://chaoss.community/?page_id=2977
* https://chaoss.community/?p=2977 (my preference)

We could include in each metric, something like this:

```
To reference this metric, please use this static URL: https://chaoss.community/?p=2977
```

**Comment by geekygirldawn** [comment]
> We could include in each metric, something like this:
> 
> ```
> To reference this metric, please use this static URL: https://chaoss.community/?p=2977
> ```

This looks perfect!

**Comment by GeorgLink** [comment]
> Should we put this near the top of the page so that people see it or would that been too much of a distraction from the metric content?

+1 for bottom

It would be too distracting at the top. I prefer to add this at the end of the page, even if it means some people would miss it and still reference the human-readable URL.

**Comment by GeorgLink** [comment]
What's the plan for the metric pages moving forward with the website redesign?

I realized we have two pages for each metric, with different page IDs:

* https://chaoss.community/metric-bus-factor/ (regular, old?)
  * https://chaoss.community/?p=2977
* https://chaoss.community/kb/bus-factor/ (knowledge base, new?)
  * https://chaoss.community/?p=3944

Ping @ElizabethN

**Comment by ElizabethN** [comment]
@klumb are you adding the stable links to each metric or am I doing that? Just wanted to clarify - I'm happy either way :blush:

**Comment by klumb** [comment]
@ElizabethN I will take care of the permalinks in WordPress as well. 

I will be adding them to the metrics release spreadsheet along with the keywords. 

Just a note about the duplicate Metrics pages. The plan was to move the pages into the KB so I am deleting the old pages but for the human-readable URL we will continue to use our current naming policy - /metric-metric-name/ Ex. https://chaoss.community/metric-labor-investment/
Is that cool with everyone?


Thanks, @geekygirldawn and @GeorgLink for continuing the discussion on human readable URLs and getting this sorted. 

We will need to do PRs to add the keywords and permalink to the metrics pages themselves. Feel free to automate that if you want otherwise, I will just do all the PRs for each WG in one go at the end. 

Permalink will probably need to be added to the metrics template along with guidance on how to add it. Am I hearing the bottom of page?

**Comment by GeorgLink** [comment]
Thank you @klumb !

+1 to keep the current URL naming policy for human-readable URL -- that is least likely to break existing links and it is more understandable than `kb/metric-name` -- if we were to change it, maybe change the first hyphen to a slash like this: `metric/metric-name` -- but better not change it at all!

+1 to bottom of the metric template

**Comment by geekygirldawn** [comment]
I agree that the bottom of the page is probably the right place for the link. However, I'm not sure about adding it to the metric template. Since we don't have the permalink until after the metric gets published in Wordpress, it isn't something that people can just fill out as part of the template. I'm guessing it will always require a second PR after the metric is published on the website, rather than something the metrics author fills out.

@klumb Let me know if you want me to regenerate the csv file after you've removed the duplicate pages with 'kb' in the link. Since my script used https://chaoss.community/kbtopic/all-metrics/ as the starting point for the urls, the csv file has the kb links. If the new links are populated from https://chaoss.community/kbtopic/all-metrics/, I can easily re-run the script to pick up the correct human readable URLs.

I'm a bit unsure about automating this - I suspect that automating it will take longer than just doing the PRs, since it's an easy change. If we do this manually, I'm wondering if we have any new contributors who might want to pick up this work? It might be a nice way to get some practice submitting PRs for someone new, especially if we split it up by WG and had a few people work on it. @ElizabethN - what do you think? Do you have anyone in mind who might want to do this?

**Comment by GeorgLink** [comment]
+1 to what Dawn said about not automating with a script.

@klumb , could we have a custom WordPress page template for metrics? The template could take care of displaying the static link and we would not have to add it to the metric in GitHub.

**Comment by klumb** [comment]
It's really about documenting the process. The template is go-to place for that. However, @geekygirldawn is right in that it will always require a second pull request. Maybe we can add it to the release checklist instead. That would resolve my concern.

**Comment by geekygirldawn** [comment]
Thanks! The release checklist sounds like the perfect place to add it!

**Comment by klumb** [comment]
Regarding automation, I just mentioned it because @geekygirldawn created a script and a CSV file.  You all can sort out how you want to do that.

**Comment by klumb** [comment]
Just to confirm,
Everyone is okay with this format for the permalinks:

https://chaoss.community/?p=123

**Comment by klumb** [comment]
@geekygirldawn and @GeorgLink 

Please review these PRs for the metric release checklist and the metric template
https://github.com/chaoss/community/pull/495
and 
https://github.com/chaoss/community/pull/496

**Comment by klumb** [comment]
Also, please be aware I am changing the language from permalink to Page Id for accuracy

**Comment by GeorgLink** [comment]
Thanks Kevin!

**Comment by samanthavenialogan** [comment]
Side note as well, if you create shortlinks using Rebrandly or YoURLs you'll be able to manage the throughway on any URL you want by changing it and the front end URL will always remain the same. That's how I maintain all of my shortlinks as well. 

When you set a shortlink it doesn't matter what happens on the back end the shortlink stays the same. So you just go in and update any shortlinks that return as 404s. 

You don't have to do this if you use the above yoast examples, but it's another option.

**Comment by klumb** [comment]
@samanthavenialogan WordPress automatically creates a page id, I am just capturing it and adding it to the metrics spreadsheet.

I believe @ElizabethN was going to add them to the metrics pages, but I am not sure what that will look like. We couldn't get consensus on adding guidance to the template. 

The permalinks (human-readable URLs) are based on the metric name. This hasn't really changed a whole lot.
